ocaml/xapi/nm.ml

```diff
@@ -89,6 +89,24 @@ let determine_ethtool_settings properties oc =
       []
   in
   let autoneg = proc "autoneg" in
+  let advertise =
+    let is_hex x =
+      try Scanf.sscanf x "0x%x%!" (fun _ -> true)
+      with Stdlib.Scanf.Scan_failure _ -> false
+    in
+    match List.assoc_opt "ethtool-advertise" oc with
```

Comment:
I think this function is not available due to the compiler version used on this branch. Do it like this: https://github.com/xapi-project/xen-api/pull/4703/files.

Reply:
You can use `List.assoc` and cover the exception in the match: `| exception Not_found -> []`

Reply:
I also realised I had 2 versions of settings, now amended